net/browserless_agent/src/browserless_agent.ts

```diff
@@ -0,0 +1,203 @@
+import { AgentFunction, AgentFunctionInfo, DefaultConfigData } from "graphai";
+
+interface BrowserlessInputs {
+  url: string;
+  text_content?: boolean;
+}
+
+interface BrowserlessParams {
+  apiKey?: string;
+  debug?: boolean;
+  throwError?: boolean;
+}
+
+type BrowserlessResult =
+  | string
+  | {
+      onError?: {
+        message: string;
+        status?: number;
+        error: any;
+      };
+    };
+
+export const browserlessAgent: AgentFunction<BrowserlessParams, BrowserlessResult, BrowserlessInputs, DefaultConfigData> = async ({ namedInputs, params }) => {
+  const { url, text_content } = namedInputs;
+  const throwError = params?.throwError ?? false;
+
+  const browserlessEndpoint = "https://chrome.browserless.io";
+  const browserlessToken = params?.apiKey ??
+    (typeof process !== "undefined" &&
+     process?.env?.BROWSERLESS_API_TOKEN || null);
+
+  // Check if API token is provided
+  if (!browserlessToken) {
+    const errorMessage = "Browserless API token is required. Please set the BROWSERLESS_API_TOKEN environment variable.";
+    throw new Error(errorMessage);
+  }
+
+  let endpoint: string;
```

Comment:
Use `const` unless it is absolutely necessary to use `let`. Using `let` increases the risk of unintended modifications later.